### PR TITLE
Taking out models dependency on User Struct

### PR DESCRIPTION
**Pull request by dscervantes**
- Taking out dependency of models User and creating a new User struct in Middleware. 
- [RHCLOUD-21483](https://issues.redhat.com/browse/RHCLOUD-21483)
Part two :Taking out models URL Params will be in a different PR

### CODE REVIEW

**Review by CodyWMitchell — APPROVED** [review]
LGTM!

File: middleware/user.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ import (
 	"github.com/redhatinsights/export-service-go/config"
 	"github.com/redhatinsights/export-service-go/errors"
 	"github.com/redhatinsights/export-service-go/logger"
-	"github.com/redhatinsights/export-service-go/models"
+	//"github.com/redhatinsights/export-service-go/models"
```
Comment (dehort) [inline middleware/user.go:17]: Please remove this line.

File: exports/exports.go [hunk 2]
```diff
@@ -42,11 +42,22 @@ func (e *Export) ExportRouter(r chi.Router) {
 	})
 }
 
+func SwitchUser(user middleware.User) models.User {
```
Comment (dehort) [inline exports/exports.go:45]: What do you think about renaming this method to something like `mapUserToModelsUser()`.  I know its long-winded, but I think it communicates its purpose a little better.

Also, capitalizing the first letter exports the method outside of the package.  I do not think we need to export this method outside of the package at the moment.

**Review by dehort — CHANGES_REQUESTED** [review]
Looking good!!

I left a couple of comments/suggestions.

**Review by CodyWMitchell — DISMISSED** [review]

**Review by dehort — APPROVED** [review]
LGTM!

**Pull request merged**